src/index.test.ts

```diff
@@ -0,0 +1,303 @@
+    id: "123",
+  });
+  t.is(result.statusCode, 200);
+  t.deepEqual(result.body, { id: "123", name: "John Doe" });
+});
+
+test("HTTPClient handles 404 errors on GET", async (t) => {
+  const error = await t.throwsAsync(
+    client.request("GET", "/users/{id}", { id: "999" })
+  );
+  t.is(error.message, '{"error":"User not found"}');
+});
+
+test("HTTPClient performs a GET request for metrics", async (t) => {
+  const result = await client.request("GET", "/metrics");
+  t.is(result.statusCode, 200);
+  t.deepEqual(result.body, { success: true });
+});
+
+test("HTTPClient handles invalid URL on GET", async (t) => {
+  const error = await t.throwsAsync(
+    client.request("GET", "/invalid/path")
+  );
+  t.is(error.message, '{"error":"Not Found"}');
+});
+
+// POST Tests
+test("HTTPClient performs a POST request with a body", async (t) => {
+  const result = await client.request(
+    "POST",
+    "/users",
+    {},
+    { name: "Jane Doe", email: "jane.doe@example.com" }
+  );
+  t.is(result.statusCode, 201);
+  t.deepEqual(result.body, {
+    id: "124",
+    name: "Jane Doe",
+    email: "jane.doe@example.com",
+  });
+});
+
+test("HTTPClient handles 400 errors on POST", async (t) => {
+  const error = await t.throwsAsync(
+    client.request("POST", "/users", {}, { name: "" })
+  );
+  t.is(error.message, '{"error":"Invalid data"}');
+});
+
+test("HTTPClient handles missing body on POST", async (t) => {
+  const error = await t.throwsAsync(
+    client.request("POST", "/users", {})
+  );
+  t.is(error.message, '{"error":"Invalid data"}');
+});
+
+test("HTTPClient handles POST request to invalid URL", async (t) => {
+  const error = await t.throwsAsync(
+    client.request("POST", "/invalid/path", {}, { name: "Test User" })
+  );
+  t.is(error.message, '{"error":"Not Found"}');
+});
+
+// PUT Tests
+test("HTTPClient performs a PUT request to update data", async (t) => {
+  const result = await client.request(
+    "PUT",
+    "/users/{id}",
+    { id: "123" },
+    { name: "Updated Name" }
+  );
+  t.is(result.statusCode, 200);
+  t.deepEqual(result.body, { id: "123", name: "Updated Name" });
+});
+
+test("HTTPClient handles 400 errors on PUT", async (t) => {
+  const error = await t.throwsAsync(
+    client.request("PUT", "/users/{id}", { id: "123" }, {})
+  );
+  t.is(error.message, '{"error":"Invalid data"}');
+});
+
+test("HTTPClient handles PUT request to a non-existent user", async (t) => {
+  const error = await t.throwsAsync(
+    client.request(
+      "PUT",
+      "/users/{id}",
+      { id: "999" },
+      { name: "Updated Name" }
+    )
+  );
+  t.is(error.message, '{"error":"User not found"}');
+});
+
+test("HTTPClient handles PUT request to invalid URL", async (t) => {
+  const error = await t.throwsAsync(
+    client.request(
+      "PUT",
+      "/invalid/path",
+      {},
+      { name: "Invalid Request" }
+    )
+  );
+  t.is(error.message, '{"error":"Not Found"}');
```

Comment:
`{"error":"Not Found"}` is not a message, it is payload.

The statusCode is important for understanding severity/meaning, eg 503 vs 500 vs 400 and is lost here.

As per other comments the error handling is a regression from v1. Can be fixed in the request logic.